test/test.cpp

```diff
@@ -260,7 +260,10 @@ bool runClientTests(const char* cmd, bool standalone, bool verbose)
 
         // output the ErrorsStream we got during the testing
         if (!result.wasSuccessful())
+        {
+            std::lock_guard<std::mutex> lock(ErrorMutex);
```

Comment:
Reading the commit message of 0c46b33e7213bf4f91d6f93ad73da9dc79718164, this is indeed used from multiple threads, so then makes sense. First this looked strange, since we clearly ask cppunit to run tests in a suite only on a single thread.